### PR TITLE
Add UDN properties and methods

### PR DESCRIPTION
**Pull request by sbahar619**
##### Short description:
Add UDN properties and methods.
## Summary by CodeRabbit

- **New Features**
	- Enhanced `UserDefinedNetwork` class with methods to evaluate network readiness and synchronization status.
	- Introduced properties to retrieve current status conditions, assess readiness, and check for synchronization errors.
  
These updates improve the monitoring and management capabilities of the `UserDefinedNetwork`.

### CODE REVIEW

File: ocp_resources/user_defined_network.py [hunk 1]
```diff
@@ -68,6 +68,74 @@ def to_dict(self) -> None:
...
+
+        Args:
+            condition (dict): A dictionary representing the condition of the UserDefinedNetwork.
+
+        Returns:
+            bool: True if the condition indicates a synchronization error, False otherwise.
+        """
+        return (
+            condition.get("reason") == cls.Condition.Reason.SYNC_ERROR
+            and condition.get("status") == cls.Condition.Status.FALSE
```
Comment (yossisegev) [inline ocp_resources/user_defined_network.py:101]: If the status is `false`, doesn't it mean that the error condition (SYNC_ERROR) you are checking here is not filled?

**Review by yossisegev — CHANGES_REQUESTED** [review]

File: ocp_resources/user_defined_network.py (hunk 1 above)
Comment (sbahar619) [reply ocp_resources/user_defined_network.py:101]: It's not mean the condition is not filled, it means that the condition is not applicable, which is correct as the condition represent sync error, a negative condition.
See k8s documentaion for more info:
https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#pod-conditions

Comment (yossisegev) [reply ocp_resources/user_defined_network.py:101]: Oh, my bad. I confused the condition's `reason` with its type. I now see that `SYNC_ERROR` is the `reason`, while the `type` is `NETWORK_READY`, which is false in this check.
Sorry for missing that.

**Review by yossisegev — APPROVED** [review]

File: ocp_resources/user_defined_network.py [hunk 2]
```diff
@@ -68,6 +68,74 @@ def to_dict(self) -> None:
...
+            and condition.get("type") == cls.Condition.Type.NETWORK_READY
+        )
+
+    @property
+    def conditions(self) -> List[Dict[str, Any]]:
```
Comment (dbasunag) [inline ocp_resources/user_defined_network.py:106]: I don't understand why this is needed. You can already access the conditions using self.instance.status.conditions.

File: ocp_resources/user_defined_network.py [hunk 3]
```diff
@@ -68,6 +68,74 @@ def to_dict(self) -> None:
...
+        Returns:
+            list: A list of status conditions associated with the UserDefinedNetwork instance.
+        """
+        # Cache instance to avoid repeated API requests
+        instance = self.instance
+        if instance.status and instance.status.conditions:
+            return instance.status.conditions
+        return []
+
+    @property
```
Comment (dbasunag) [inline ocp_resources/user_defined_network.py:119]: This PR seems better suited for cnv-tests, no?

Comment (myakove) [reply ocp_resources/user_defined_network.py:119]: Yes!

**Review by myakove — CHANGES_REQUESTED** [review]

File: ocp_resources/user_defined_network.py (hunk 3 above)
Comment (sbahar619) [reply ocp_resources/user_defined_network.py:119]: Got it.

**Pull request closed without merging**